app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/IngredientsProductFragment.java

```diff
@@ -73,10 +74,11 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         }
 
         if(mState != null && product.getIngredientsText() != null) {
-            SpannableStringBuilder txtIngredients = new SpannableStringBuilder(Html.fromHtml(product.getIngredientsText().replace("_","")));
-            txtIngredients = setSpanBoldBetweenTokens(txtIngredients);
-            if(!txtIngredients.toString().substring(txtIngredients.toString().indexOf(":")).trim().isEmpty()) {
-                ingredientsProduct.setText(txtIngredients);
+            //TODO The API doesn't return ingredients text with the _ token. the replace method could be removed
```

Comment:
why not just print the ingredients list (which is already available from the ingredients json fields) and print just below the allergens list ?